client-java/src/main/java/uk/gov/gchq/palisade/client/internal/resource/ResourceMessage.java

```diff
@@ -13,54 +13,42 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package uk.gov.gchq.palisade.client.resource;
+package uk.gov.gchq.palisade.client.internal.resource;
 
 import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
 import com.fasterxml.jackson.databind.annotation.JsonSerialize;
 import org.immutables.value.Value;
 
-import uk.gov.gchq.palisade.client.util.ImmutableStyle;
-
-import java.io.Serializable;
-import java.util.Map;
-import java.util.function.UnaryOperator;
+import uk.gov.gchq.palisade.client.internal.resource.WebSocketListener.Item;
 
 /**
- * A {@code Resource} object is received in a {@link Message} of type
+ * A {@code Resource} object is received in a {@link Item} of type
  * {@code MessageType#RESOURCE} after it has sent a message of type
  * {@code MessageType#CTS}.
  * <p>
  * Note that the {@code UserId} class is created at compile time. The way in
  * which the class is created is determined by the {@code ImmutableStyle}. This
  * class is also compatible with Jackson.
  *
- * @see ResourceClient
+ * @see WebSocketClient
  * @see "https://immutables.github.io/style.html"
  * @since 0.5.0
  */
 @Value.Immutable
-@ImmutableStyle
-@JsonDeserialize(as = Resource.class)
-@JsonSerialize(as = Resource.class)
-public interface IResource extends Serializable {
+@JsonDeserialize(as = ImmutableResourceMessage.class)
+@JsonSerialize(as = ImmutableResourceMessage.class)
```

Comment:
That's there so that testing is far easier. It allows creating the object and then serialising it to JSON without having to manually create and embed the JSON in the test which is error prone with all the escaping required. This would be so much easy with multi-line strings.